cleanup: use `rbd.Manager` for fetching volumes in CSI-Addons operations

nixpanic opened the pull request:
ReclaimSpace and EncryptionKeyRotation now use the `Volume` type instead of the (supposed to be) internal `rbdVolume`.

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

* `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)

</details>

nixpanic committed:
rbd: add sub-types for large Volume type

Introduce `snapshottableVolume` and `csiAddonsVolume` types which group
related functions together.

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic committed:
csiaddons: use `rbd.Manager` for encryption key rotation

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic committed:
csiaddons: use `rbd.Manager` within `ReclaimSpaceControllerServer`

Signed-off-by: Niels de Vos <ndevos@ibm.com>

Rakshith-R commented on internal/rbd/types/volume.go:
```diff
@@ -25,9 +25,48 @@ import (
+	// if the flattenMode is FlattenModeForce, it flattens the image itself.
+	// if the parent image is in trash, it returns an error.
+	// if the parent image exists and is not enabled for mirroring, it returns an error.
+	HandleParentImageExistence(ctx context.Context, flattenMode FlattenMode) error
```
This is required for csi addons replication enable operation & group add operation only and not for CSI spec calls.

Rakshith-R reviewed (dismissed):
LGTM, thanks !

Madhu-1 commented:
@mergifyio queue

iPraveenParihar commented on internal/csi-addons/rbd/encryptionkeyrotation.go:
```diff
@@ -32,11 +32,15 @@ import (
 
 type EncryptionKeyRotationServer struct {
 	*ekr.UnimplementedEncryptionKeyRotationControllerServer
+	driver  string
```
this is `csiID`, naming it as driver causes confusion to me. And also everywhere else we have `csiID` naming lets stick to that?

nixpanic replied:
`csiID` is a very strangely called variable. In the CSI Specification it is the `name` returned in the [IdentityService.GetPluginInfo operation](https://github.com/container-storage-interface/spec/blob/master/spec.md#getplugininfo). It is not so much an ID, it is more the name of the driver (for its current deployment). Even in Kubernetes `name` is what is used, see `deploy/rbd/kubernetes/csidriver.yaml`.

`csiID` is also used for variables of the `util.CSIIdentifier` type, and that is why I prefer a different variable name for the driver-instance.

I'll send a cleanup to rename the `csiID` variable for the driver-instance, and will call it `driverName` for clarity.

nixpanic replied:
Oh, wait, in most cases it is actually the `instanceID` for the driver to allow multi-tenancy on the same Ceph cluster (it is used as the location of the journal where Ceph-CSI stores its metadata). I'll call it `driverInstance` instead.

iPraveenParihar commented on internal/csi-addons/rbd/reclaimspace.go:
```diff
@@ -37,13 +37,18 @@ import (
 type ReclaimSpaceControllerServer struct {
 	*rs.UnimplementedReclaimSpaceControllerServer
+
+	driver      string
```
here as well

iPraveenParihar reviewed (changes requested):
requested changes

nixpanic committed:
cleanup: rename `csiID` to `driverInstance`

The attribute and variable `csiID` ise used for at least two different
things:

 - name of the driver instance, used for journalling metadata
 - objects of the CSIIdentifier struct, composing a volume-handle

By changing the name of the `csiID` attribute for driver instances to
`driverInstance`, any confusion should be prevented.

Signed-off-by: Niels de Vos <ndevos@ibm.com>

iPraveenParihar reviewed (approved):
LGTM. Thanks!

Madhu-1 commented:
@mergifyio queue
